File: exercise/php/day15/src/SantaChristmasList/Dependencies/Factory.php
<?php

namespace SantaChristmasList\Dependencies;

use SantaChristmasList\Models\Gift;
use SantaChristmasList\Models\ManufacturedGift;

class Factory extends \ArrayObject
{
    public function findManufacturedGift(Gift $gift): ?ManufacturedGift
    {
        return $this->offsetExists($gift->getName()) ? $this->offsetGet($gift->getName()) : null;
    }

    public function set(Gift $gift, ManufacturedGift $manufacturedGift): void
    {
        $this->offsetSet($gift->getName(), $manufacturedGift);
    }
}

File: exercise/java/day15/src/main/java/santaChristmasList/operations/dependencies/Factory.java
package santaChristmasList.operations.dependencies;

import santaChristmasList.operations.models.Gift;
import santaChristmasList.operations.models.ManufacturedGift;

import java.util.HashMap;

public class Factory extends HashMap<Gift, ManufacturedGift> {
    public ManufacturedGift findManufacturedGift(Gift gift) {
        return this.getOrDefault(gift, null);
    }
}